drivers: sensor: bmc150_magn: Update driver to use DEVICE_DT_GET

benjaminbjornsson opened the pull request.

benjaminbjornsson committed:
drivers: sensor: bmc150_magn: Update driver to use i2c_dt_spec

Simplify driver by using i2c_dt_spec for bus access.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
drivers: sensor: bmc150_magn: Update driver to use gpio_dt_spec

Simplify driver by using gpio_dt_spec for bus access.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

galak reviewed (changes requested):
LGTM.  Can you update `samples/sensor/magn_polling` as well.

benjaminbjornsson commented:
> LGTM. Can you update `samples/sensor/magn_polling` as well.

Of course!

benjaminbjornsson committed:
samples: sensor: bmc150magn: Convert sample to use DEVICE_DT_GET_ONE

Move to use DEVICE_DT_GET_ONE instead of device_get_binding as
we work on phasing out use of DTS 'label' property.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

gmarull commented on samples/sensor/bmc150_magn/src/main.c:
```diff
@@ -10,10 +10,17 @@
+void main(void)
 {
-	int ret;
+	struct device *dev;
 	struct sensor_value value_x, value_y, value_z;
+	int ret;
+
+	dev = DEVICE_DT_GET_ONE(bmc150_magn);
+	if (!device_is_ready(dev)) {
+		printk("sensor: device not ready.\n");
+		return;
+	}
```
```suggestion
	const struct device *dev = DEVICE_DT_GET_ONE(bmc150_magn);
	struct sensor_value value_x, value_y, value_z;
	int ret;

	if (!device_is_ready(dev)) {
		printk("sensor: device not ready.\n");
		return;
	}
```

MaureenHelm reviewed (changes requested):
I'd prefer not to rename the sample. It's close to being generic for any magnetometer and I don't want to keep proliferating driver-specific sensor samples.

benjaminbjornsson commented:
> I'd prefer not to rename the sample. It's close to being generic for any magnetometer and I don't want to keep proliferating driver-specific sensor samples.

Changed it back!

gmarull commented on samples/sensor/magn_polling/src/main.c:
```diff
@@ -10,10 +10,16 @@
+void main(void)
 {
-	int ret;
+	struct device *dev = DEVICE_DT_GET_ONE(bmc150_magn);
```
note: this sample is not generic anyway, we should likely rely on aliases, otherwise it's specific to `bmc150_mgn`, which btw should be `bosch_bmc150_magn`. does it compile now?

benjaminbjornsson replied:
Should we come up with an alias now or is that for later?
Actually it didn't compile... What's the best way to compiling a sample? Using west build I guess I'm stuck to either finding a board with the sensor in the DTS already or just adding a temporary app.overlay in the sample directory?

gmarull replied:
You can use `twister`, this way you'll make use of the information in `sample.yaml`. I've always found weird that `west` doesn't make use of `sample.yaml` to tell you _hey, the board you're building for is not compatible!_, we can probably improve that for a better user experience.

benjaminbjornsson replied:
Thanks, I should definitely take a closer look at twister.
